### PR TITLE
*: replaced ignition and cloud config with container linux configs

### PR DESCRIPTION
**Pull request by dgonyeo**
Here's an incomplete pass at changing all the OS docs over to using container linux configs instead of ignition or cloud configs.

Everything alphabetically before `iscsi.md` should be good to go, with the exception of `cluster-architecture.md`, which needs a complete rewrite.

### CODE REVIEW

**Comment by dgonyeo** [comment]
This should now be complete, and ready for review

File: os/customizing-sshd.md [hunk 1]
```diff
@@ -1,57 +1,56 @@
...
+        inline: |
+          # Use most defaults for sshd configuration.
+          UsePrivilegeSeparation sandbox
+          Subsystem sftp internal-sftp
 
-      PermitRootLogin no
-      AllowUsers core
-      PasswordAuthentication no
-      ChallengeResponseAuthentication no
+          PermitRootLogin no
```
Comment (dgonyeo) [inline os/customizing-sshd.md]: You mean to inlcude the default config and append lines to it? `Include` in what context? ct/ignition don't have that functionality, and I don't see an `Include` directive in https://linux.die.net/man/5/sshd_config.

**Comment by dgonyeo** [comment]
Found and replaced the ignition and cloud config references outside of the `os` dir, which I had forgotten about initially. Need to rewrite large parts of `etcd/tls-etcd-clients.md`. Going to do the standard CLC thing you mentioned, still getting to it.

**Comment by dgonyeo** [comment]
Just did what I probably should've done in the beginning: pulled in [ct's docs test](https://github.com/coreos/container-linux-config-transpiler/blob/master/doc/doc_test.go) and tweaked it for here. All markdown files in this repo are now parsed for CLC blocks, and the found blocks are run through the parser and converter. Any warnings or errors cause the test to fail.

There's no CI for it on this repo, but at the very least it's something that docs authors can run when they modify/add CLC blocks.

File: os/mounting-storage.md [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
+Mount units name the source filesystem and target mount point, and optionally the filesystem type. *Systemd* mounts filesystems defined in such units at boot time. The following example mounts an [EC2 ephemeral disk](booting-on-ec2.md#instance-storage) at the node's `/media/ephemeral` directory, and is therefore named `media-ephemeral.mount`:
 
-coreos:
+```container-linux-config
+systemd:
   units:
     - name: media-ephemeral.mount
-      command: start
-      content: |
+      enable: true
```
Comment (crawford) [inline os/mounting-storage.md]: There is no install section.

File: os/notes-for-distributors.md [hunk 3]
```diff
@@ -18,9 +18,30 @@ gpg --verify coreos_production_openstack_image.img.bz2.sig
...
+There are two predominant ways that a Container Linux image can be easily customized for a specific operating environment: through Ignition, a first-boot provisioning tool that runs during a machine's boot process, and through [cloud-config](https://github.com/coreos/coreos-cloudinit/blob/master/Documentation/cloud-config.md), an older tool that runs every time a machine boots.
 
-Use cloud-config to handle platform specific configuration such as custom networking, running an agent on the machine or injecting files onto disk. Container Linux will automatically parse and execute `/usr/share/oem/cloud-config.yml` if it exists. Your cloud-config should create additional units that process user-provided metadata, as described below.
+### Ignition
+
+[Ignition][ignition] is a tool that acquires a JSON config file when a machine first boots, and uses this config to perform tasks such as formatting disks, creating files, modifying and creating users, and adding systemd units. How Ignition acquires this config file varies per-platform, and it is highly recommended that providers ensure Ignition has [support for their platform][ign-platforms].
```
Comment (crawford) [inline os/notes-for-distributors.md]: The method by which Ignition acquires this config varies per platform.

File: os/notes-for-distributors.md [hunk 4]
```diff
@@ -18,9 +18,30 @@ gpg --verify coreos_production_openstack_image.img.bz2.sig
...
+
+Ignition has a default config for each supported platform that is appended to any user provided config. Use Ignition to handle platform specific configuration such as custom networking, running an agent on the machine, or injecting files onto disk.
```
Comment (crawford) [inline os/notes-for-distributors.md]: This first statement is an implementation detail and will be removed eventually.

File: os/other-settings.md [hunk 5]
```diff
@@ -8,93 +8,42 @@ Most Linux kernel modules get automatically loaded as-needed but there are a som
...
+      contents:
+        inline: nf_conntrack
 ```
 
 ### Loading kernel modules with options
 
-The following section demonstrate how to provide module options when
-loading. After these configs are processed, the dummy module is loaded
-into the kernel, and five dummy interfaces are added to the network stack.
+The following section demonstrate how to provide module options when loading. After these configs are processed, the dummy module is loaded into the kernel, and five dummy interfaces are added to the network stack.
```
Comment (crawford) [inline os/other-settings.md]: demonstrates

File: os/booting-on-openstack.md [hunk 6]
```diff
@@ -72,41 +72,37 @@ $ glance image-create --name Container-Linux \
...
-    listen-peer-urls: http://$private_ipv4:2380
-  units:
-    - name: etcd2.service
-      command: start
-    - name: fleet.service
-      command: start
-ssh_authorized_keys:
-  # include one or more SSH public keys
-  - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDGdByTgSVHq.......
+```container-linux-config:openstack-metadata
```
Comment (dgonyeo) [inline os/booting-on-openstack.md]: The line immediately after this block reads:
> The `{PRIVATE_IPV4}` and `{PUBLIC_IPV4}` substitution variables are fully supported in cloud-config on OpenStack deployments using the metadata service. Unfortunately systems relying on config drive are currently unsupported.

Is that sufficient?

Comment (crawford) [inline os/booting-on-openstack.md]: This isn't cloud-config though.

**Comment by dgonyeo** [comment]
Just pushed some stuff:
- fixed errors in the ec2 CLC
- updated the digitalocean screenshot to be of an ignition config instead of a cloud config
- finished the `etcd/tls-etcd-clients.md` file
- switched all ignition-supported cloud providers to using the same, heavily commented CLC that starts etcd. This relies on https://github.com/coreos/container-linux-config-transpiler/pull/54, which is not yet in a release. I just tested booting a machine in ec2 with this CLC, and it behaved as expected.

File: os/customizing-sshd.md [hunk 7]
```diff
@@ -1,57 +1,56 @@
...
-  - name: sshd.socket
-    command: restart
-    runtime: true
-    content: |
-      [Socket]
-      ListenStream=2222
-      FreeBind=true
-      Accept=yes
+    - name: sshd.socket
+      enable: true
```
Comment (crawford) [inline os/customizing-sshd.md]: No install section.

File: os/mounting-storage.md [hunk 8]
```diff
@@ -66,13 +64,12 @@ Container Linux [561.0.0](https://coreos.com/releases/#561.0.0) and later are in
...
-```cloud-config
-#cloud-config
-coreos:
+```container-linux-config
+systemd:
   units:
     - name: format-var-lib-docker.service
-      command: start
-      content: |
+      enable: true
```
Comment (crawford) [inline os/mounting-storage.md]: No install section.

File: os/power-management.md [hunk 9]
```diff
@@ -21,17 +21,14 @@ modprobe cpufreq_conservative
...
-
-coreos:
+```container-linux-config
+systemd:
   units:
     - name: cpu-governor.service
-      command: start
-      runtime: true
-      content: |
+      enable: true
```
Comment (crawford) [inline os/power-management.md]: No install section.

File: os/scheduling-tasks-with-systemd-timers.md [hunk 10]
```diff
@@ -39,56 +39,31 @@ You can also create timer with different name, i.e. `task.timer`. In this case y
...
         [Unit]
         Description=Prints date into /tmp/date file
 
         [Service]
         Type=oneshot
         ExecStart=/usr/bin/sh -c '/usr/bin/date >> /tmp/date'
     - name: date.timer
-      command: start
-      content: |
+      enable: true
```
Comment (crawford) [inline os/scheduling-tasks-with-systemd-timers.md]: No install section.

File: etcd/etcd-live-cluster-reconfiguration.md [hunk 11]
```diff
@@ -4,29 +4,30 @@ This document describes the reconfiguration or recovery of an etcd cluster runni
...
-coreos:
-  etcd2:
-    advertise-client-urls: http://<PEER_ADDRESS>:2379
-    initial-advertise-peer-urls: http://<PEER_ADDRESS>:2380
-    listen-client-urls: http://0.0.0.0:2379,http://0.0.0.0:4001
-    listen-peer-urls: http://0.0.0.0:2380
-    discovery: https://discovery.etcd.io/<token>
+```container-linux-config
+etcd:
+  version: 3.1.4
```
Comment (crawford) [inline etcd/etcd-live-cluster-reconfiguration.md]: We were going to toss the version, right?

File: os/quickstart.md [hunk 12]
```diff
@@ -40,29 +40,22 @@ CoreOS (beta)
...
-      command: start
-    - name: fleet.service
-      command: start
-  etcd2:
-    # generate a new token for each unique cluster from https://discovery.etcd.io/new?size=3
-    # specify the initial size of your cluster with ?size=X
-    discovery: https://discovery.etcd.io/<token>
+```container-linux-config
+etcd:
+  version: 3.1.4
```
Comment (crawford) [inline os/quickstart.md]: Version.

**Pull request merged**